### PR TITLE
doc: added new document to list Node-API bindings.

### PR DESCRIPTION
**Pull request by NickNaso**
This PR add new document to list Node-API bindings.
Refs:
- https://github.com/nodejs/abi-stable-node/issues/439

### CODE REVIEW

File: node-api.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+|[go-node-api](https://github.com/napi-bindings/go-node-api)| Go|
+
+<a name="#runtimes"></a>
+
+## Node-API bindings for other runtimes
+
+|Project | Programming language|
+|--------|---------------------|
+|[bun](https://github.com/oven-sh/bun)| Zig|
+|[deno](https://github.com/denoland/deno)| Rust|
```
Comment (legendecas) [inline node-api.md]: Should we add words on no endorsement by the listed runtimes and organizations, and no affiliation with the organizations?

**Comment by KevinEady** [comment]
We discussed this in the 11 Nov Node API meeting.

Since this file is only relating to engine bindings, would it be better to name the file something like `node-api-engine-bindings.md`, since we are planning on linking to this page within the core Node.js documentation page?

File: node-api.md (hunk 1 above)
Comment (mhdawson) [reply node-api.md]: I wonder if electron should be included in the list as well?

File: node-api.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+There are many projects that implement Node-API bindings and allow to develop
+a native addon for Node.js, but at the same time  Node-API has been implemented
+in other JavaScript/TypeScript runtimes. In the following section there are
+some of them.
+
+- [Node-API bindings for other languages](#languages)
+- [Node-API bindings for other runtimes](#runtimes)
+
+<a name="#languages"></a>
```
Comment (mhdawson) [inline node-api.md]:
``` suggestion
<a name="#languages"></a>
**NOTE** The Node-api team has not tried out or validated the projects listed
below and does not endorse them in any way. Similarly, being listed below does
not indicate that those projects endorse Node.js or Node-api.
The lists are simply provided as a starting point for you to do your own evaluation.
```

**Review by mhdawson — APPROVED** [review]
LGTM

**Review by legendecas — APPROVED** [review]

**Pull request merged**